From 814d3471b0e1effc894b70d5fa1599479b48fbc5 Mon Sep 17 00:00:00 2001
From: Damjan Jovanovic <damjan@apache.org>
Date: Sat, 13 Oct 2012 10:11:16 +0000
Subject: [PATCH] Fix setExifGPSTag().

Submitted by: Herve M. <hmarmy at gmail dot com>
Jira issue key: IMAGING-92



git-svn-id: https://svn.apache.org/repos/asf/commons/proper/imaging/trunk@1397808 13f79535-47bb-0310-9956-ffa450edef68
---
 src/changes/changes.xml                                     | 3 +++
 .../commons/imaging/formats/tiff/write/TiffOutputSet.java   | 6 +++---
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index 925d8bb9c..a1ac36901 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -230,6 +230,9 @@ The <action> type attribute can be add,update,fix,remove.
       <action issue="IMAGING-89" dev="damjan" type="add">
         Detect the YCCK color space/type in JPEG files' APP14 segment.
       </action>
+      <action issue="IMAGING-92" dev="damjan" type="fix" due-to="hmarmy">
+        Issue trying to update GPS metadata info in a JPG.
+      </action>
     </release>
     
   </body>
diff --git a/src/main/java/org/apache/commons/imaging/formats/tiff/write/TiffOutputSet.java b/src/main/java/org/apache/commons/imaging/formats/tiff/write/TiffOutputSet.java
index c7b3cdc10..14aa5281b 100644
--- a/src/main/java/org/apache/commons/imaging/formats/tiff/write/TiffOutputSet.java
+++ b/src/main/java/org/apache/commons/imaging/formats/tiff/write/TiffOutputSet.java
@@ -144,11 +144,11 @@ public void setGPSInDegrees(double longitude, double latitude)
         latitude = Math.abs(latitude);
 
         gpsDirectory.removeField(GpsTagConstants.GPS_TAG_GPS_LONGITUDE_REF);
-        gpsDirectory.add(GpsTagConstants.GPS_TAG_GPS_DEST_LONGITUDE_REF,
+        gpsDirectory.add(GpsTagConstants.GPS_TAG_GPS_LONGITUDE_REF,
                 longitudeRef);
 
         gpsDirectory.removeField(GpsTagConstants.GPS_TAG_GPS_LATITUDE_REF);
-        gpsDirectory.add(GpsTagConstants.GPS_TAG_GPS_DEST_LATITUDE_REF,
+        gpsDirectory.add(GpsTagConstants.GPS_TAG_GPS_LATITUDE_REF,
                 latitudeRef);
 
         {
@@ -163,7 +163,7 @@ public void setGPSInDegrees(double longitude, double latitude)
 
             gpsDirectory.removeField(GpsTagConstants.GPS_TAG_GPS_LONGITUDE);
             gpsDirectory
-                    .add(GpsTagConstants.GPS_TAG_GPS_DEST_LONGITUDE,
+                    .add(GpsTagConstants.GPS_TAG_GPS_LONGITUDE,
                             RationalNumberUtilities
                                     .getRationalNumber(longitudeDegrees),
                             RationalNumberUtilities
